python/tests/test_extend.py

```diff
@@ -20,3 +23,29 @@ def test_extend(self):
               metta)
 
 
+class ExtendGlobalTest(unittest.TestCase):
+
+    def test_extend_global(self):
+        '''
+        This test is intended to check that if the extension uses internal states,
+        they are not lost between metta.run's and are also accessible from Python
+        '''
+        from extension import g_object
+        # Sanity check
+        self.assertEqual(g_object, None)
+        metta = MeTTa()
+        metta.run('''
+          !(extend-py! extension)
+          !(set-global! 42)
+        ''')
+        # Checking that the object is accessible and its value is correct
+        self.assertEqual(
+          metta.run('!(get-global)'),
+          metta.run('! 42'))
+        # We may have to reimport g_object, but its value should be set
+        from extension import g_object
```

Comment:
Minor but we also can use `import` with `extension.g_object` instead of `from` to make it work without re-importing:
```python
        import extension
        # Sanity check
        self.assertEqual(extension.g_object, None)
        metta = MeTTa()
        metta.run('''
          !(extend-py! extension)
          !(set-global! 42)
        ''')
        # Checking that the object is accessible and its value is correct
        self.assertEqual(
          metta.run('!(get-global)'),
          metta.run('! 42'))
        self.assertEqual(extension.g_object, 42)
```

Reply:
I've added it as a comment because the remark about re-importing can also be helpful